### ISSUE TITLE
Buffer previewer (?) affects WinOpts

### ISSUE THREAD

**Issue by savchenko** [linrongbin16/fzfx.nvim#603]
Running [buffer selector](https://github.com/linrongbin16/fzfx.nvim/wiki/Helpful-snippets#buffer-selector) using v5.1.6 creates the following window:

![image](https://github.com/linrongbin16/fzfx.nvim/assets/300146/bbb8ac1b-b3ab-4ef8-a1d9-9781e493e45a)

In the subsequent versions, it returns this:

![image](https://github.com/linrongbin16/fzfx.nvim/assets/300146/0b5b2b0e-1cfc-418a-acab-f3b2e75772f9)

Is this a bug or a change of spec?

**Comment by linrongbin16** [comment]
> Running [buffer selector](https://github.com/linrongbin16/fzfx.nvim/wiki/Helpful-snippets#buffer-selector) using v5.1.6 creates the following window:
> 
> ![image](https://private-user-images.githubusercontent.com/300146/304563613-bbb8ac1b-b3ab-4ef8-a1d9-9781e493e45a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4Njc3OTQsIm5iZiI6MTcwNzg2NzQ5NCwicGF0aCI6Ii8zMDAxNDYvMzA0NTYzNjEzLWJiYjhhYzFiLWIzYWItNGVmOC1hMWQ5LTk3ODFlNDkzZTQ1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QyMzM4MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OGU4MjgyN2YzZDgxZjhjMDU1NzdiYmRkNThjZTg4YWNiMjRjYzE3ZGIwZjVkZjQ4YTdjYzA5ODMwNjA0YTBhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RVwvCc9jYt39xH0dzmWMqL6NzL4xM-LkCs2kYPEln94)
> 
> In the subsequent versions, it returns this:
> 
> ![image](https://private-user-images.githubusercontent.com/300146/304563862-0b5b2b0e-1cfc-418a-acab-f3b2e75772f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4Njc3OTQsIm5iZiI6MTcwNzg2NzQ5NCwicGF0aCI6Ii8zMDAxNDYvMzA0NTYzODYyLTBiNWIyYjBlLTFjZmMtNDE4YS1hY2FiLWYzYjJlNzU3NzJmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIxM1QyMzM4MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMjllODc1ODY3NTgwNDllYzI4MmM1OWJkZDhhMjhlZDY5NjQ0ZTdmOGViYmUyY2VkYTY5MzQ0ZmI1OWMwNzFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Elv6WPj6R3WsDB8RwsdCdZbFmtyFsbATdU_h-_SlmF4)
> 
> Is this a bug or a change of spec?

should be a bug, I will take a look

**Comment by linrongbin16** [comment]
hi @savchenko,

please set `let g:fzfx_disable_buffer_previewer=1` to disable the buffer previewer as a workaround.

I will fix this issue.

**Comment by linrongbin16** [comment]
hi @savchenko , 

Can I reproduce this issue with the `buffers` configuration you post here: https://github.com/linrongbin16/fzfx.nvim/wiki/Helpful-snippets#buffer-selector ?

**Comment by linrongbin16** [comment]
hi @savchenko , thanks to your report & help, #609 has finally fixed this issue, please pull main branch and try again!

**Comment by linrongbin16** [comment]
hi @savchenko ,

I found the config you post here: https://github.com/linrongbin16/fzfx.nvim/wiki/Advanced-Configuration#buffer-selector, it looks very small when I am trying it.

Are you sure it's really working in this way?

**Comment by linrongbin16** [comment]
it could be a bug, because I did many hacks on fzf options, to make it compatible with nvim buffer previewer.

I will take a look at this issue.

**Comment by linrongbin16** [comment]
Update:

I found the root cause, there're 2 things need to note:

1. should set fzf options to lua list:

   ```lua
   fzf_opts = {
     '--info=hidden',
     '--padding=0',
     '--header=',
     '--preview-window=hidden'
   }
   ```
   This will help this plugin to correctly parse fzf options.

2. The `'--preview-window=hidden'` option is however been ignored when starting `FzfxBuffers` command, I'm working on it.

**Comment by linrongbin16** [comment]
hi @savchenko ,

The 2nd issue mentioned in above comments is fixed in #689 , please have a try!
